package mvp.pipeline;

import "mvp/Pipeline/AlgorithmSettings.proto";

message SessionDesc {
  message Input {
    required string image_pattern = 1;
    required string camera_pattern = 2;
  }

  message Render {
    required string bbox = 1;
    optional int32 level = 2;
  }

  message Output {
    required string platefile = 1;
    required string datum = 2;
    optional string map_projection = 3 [default="equi"];
    optional int32 tile_size = 4 [default = 64];
  }

  required Input input = 1;
  required Render render = 2;
  required Output output = 3;
  required AlgorithmSettings algorithm_settings = 4;
}
